### PR TITLE
Simulation - Correct TrackIds in MCTrack and TrackRefs in subevent merging

### PR DESCRIPTION
**Pull request by amorsch**
solves JIRA O2-1174
(+ some automatic clang style changes)

### CODE REVIEW

**Comment by sawenzel** [comment]
The unit test that checks the kinematics is now failing. I'll need to take a closer look into this. I thought that trackID handling after merge was implemented correctly. Do you have an example where this is wrong?

(It appears that this PR contains commits already merged.)

**Comment by amorsch** [comment]
For the TrackRefs is is even written in the comments that it is missing and it is clear from code that this is not done anywhere. The same for the trackmothers, they are only copied without any update.
Please tell me where it fails and I'll fix it.

**Comment by amorsch** [comment]
---->
 // TODO: fix track numbers in TrackRefs
    merge<std::vector<o2::TrackReference>>("TrackRefs", *tree, *mOutTree);
    merge<o2::dataformats::MCTruthContainer<o2::TrackReference>>("IndexedTrackRefs",
                                                                 *tree, *mOutTree);
<-----
I accidentally also deleted the second line for the IndexedTrackRefs.
That's why the test fails. Sorry for this.
Do they also used track indices?
Andreas

**Comment by sawenzel** [comment]
Ok. I'll take a closer look.

File: run/O2HitMerger.h [hunk 1]
```diff
@@ -315,7 +317,54 @@ class O2HitMerger : public FairMQDevice
   {
     to.mergeAtBack(from);
   }
-
+  template <typename T>
+  void remapTrackIdsAndMerge(std::string brname, TTree& origin, TTree& target, std::vector<int> trackoffsets)
+  {
```
Comment (sawenzel) [inline run/O2HitMerger.h]: please use `const std::vector<int>& trackoffsets`

File: run/O2HitMerger.h [hunk 2]
```diff
@@ -424,11 +473,10 @@ class O2HitMerger : public FairMQDevice
...
     // b) merge the general data
-    merge<std::vector<o2::MCTrack>>("MCTrack", *tree, *mOutTree);
-    // TODO: fix track numbers in TrackRefs
-    merge<std::vector<o2::TrackReference>>("TrackRefs", *tree, *mOutTree);
-    merge<o2::dataformats::MCTruthContainer<o2::TrackReference>>("IndexedTrackRefs",
-                                                                 *tree, *mOutTree);
+    //
+    // for MCTrack remap the motherIds and merge at the samee go
+    remapTrackIdsAndMerge<std::vector<o2::MCTrack>>("MCTrack", *tree, *mOutTree, trackoffsets);
+    remapTrackIdsAndMerge<std::vector<o2::TrackReference>>("TrackRefs", *tree, *mOutTree, trackoffsets);
```
Comment (sawenzel) [inline run/O2HitMerger.h]: we need to keep the `o2::dataformats::MCTruthContainer` for now to not fail the test. In the end we can decide which format to keep. This one here allows to easiky query TrackRefs given a trackID.

**Comment by sawenzel** [comment]
We can merge this when the small comments are resolved. Once merged, I will unify the treatment of the two versions of TrackRef containers.

**Pull request merged**